### ISSUE TITLE
Prometheus: Datasource variable causes stale label-browser-data.

### ISSUE THREAD

**Issue by axelavargas** [grafana/grafana#52432]
**What happened**:

Issue found when triaging: #51355 

Editing a panel that uses a Datasource variable causes stale label-browser-data.

**What you expected to happen**:
It should keep the labels browser data up to date

**How to reproduce it (as minimally and precisely as possible)**:

1. create two Prometheus data sources
2. in a dashboard create a variable type=data_source
3. create a dashboard panel with the data-source being the newly created variable
4. if you start switching the datasource-variable-value, the label-browser in the query-editor does not update.

https://user-images.githubusercontent.com/239999/178938974-1c9782a8-2f67-401e-ac8d-43c2b9d336ea.mp4

**Anything else we need to know?**:

**Environment**:
- Grafana version:  I tested in Grafana 8.1.2 and Grafana 9.0.2
- Data source type & version: Prometheus

**Comment by aocenas** [comment]
Yeah we probably don't trigger a refresh of the values when the data source changes, this probably only happens in this way in a dashboard with a dashboard variable otherwise I assume the component gets unmounted.

**Comment by gtk-grafana** [comment]
@grafana/observability-metrics Needs to investigate further and determine the effort for a solution.

**Comment by jcolladokuri** [comment]
Hi @axelavargas thank you for reporting, I took a look at this and it seems like it has been fixed! I will go ahead and close this issue, but  please feel free to reopen it if you are still running into any problems!

Thanks!
